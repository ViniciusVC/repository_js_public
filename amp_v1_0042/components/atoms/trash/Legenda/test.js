import React from "react";
import { shallow } from "enzyme";
import Component from "./index";
import { exec } from "child_process";

describe("Legenda >", () => {
  it("deve renderizar com o tema default, recebendo as props required", () => {
    const wrapper = shallow(
      <Component franquia={70} consumo={5} dataRenova="14/03/2015" />
    );
    expect(
      wrapper
        .find("strong")
        .first()
        .text()
    ).toContain("3.5");
    expect(
      wrapper
        .find("strong")
        .at(1)
        .text()
    ).toContain("66.5");
    expect(
      wrapper
        .find("Muted")
        .children()
        .at(1)
        .text()
    ).toContain("14/03/2015");
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
