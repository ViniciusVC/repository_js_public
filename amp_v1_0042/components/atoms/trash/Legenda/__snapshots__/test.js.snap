// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legenda > deve renderizar com o tema default, recebendo as props required 1`] = `
<small>
  <Row
    align={null}
    justify={null}
    margin={null}
    padding={null}
    theme={
      Object {
        "flex": "
  position: relative;
  display: flex !important;
  flex-direction: column;
  ",
      }
    }
  >
    <Col
      align={null}
      justify={null}
      margin={null}
      padding={null}
      theme={
        Object {
          "flex": "
  position: relative;
  display: flex !important;
  flex-direction: column;
  ",
        }
      }
    >
      <Legend
        theme={
          Object {
            "gradient": "linear-gradient(to right, #ff6d00, #d82482)",
            "primary": "#d82482",
            "secondary": "#909090",
          }
        }
      >
        Consumido: 
        <strong>
          3.5
           GB  [
          5
          %]
        </strong>
      </Legend>
      <Legend
        theme={
          Object {
            "gradient": "linear-gradient(to right, #ff6d00, #d82482)",
            "primary": "#d82482",
            "secondary": "#909090",
          }
        }
      >
        Disponivel: 
        <strong>
          66.5
           GB  [
          95
          %]
        </strong>
      </Legend>
    </Col>
    <Muted
      theme={
        Object {
          "secondary": "#909090",
        }
      }
    >
      Data de renovação: 
      14/03/2015
    </Muted>
  </Row>
</small>
`;
